pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -779,25 +774,33 @@ func configureDhcpConfigV6Addresses(b *MasqueradeBindMechanism, err error) error
 		b.vmIPv6NetworkCIDR = api.DefaultVMIpv6CIDR
 	}
 
-	defaultGatewayIpv6, vmIpv6, err := b.handler.GetHostAndGwAddressesFromCIDR(b.vmIPv6NetworkCIDR)
+	gatewayAddr, vmAddr, err := b.generateGatewayAndVmIPAddrs(b.vmIPv6NetworkCIDR)
 	if err != nil {
-		log.Log.Reason(err).Errorf("failed to get gw and vm available ipv6 addresses from CIDR %s", b.vmIPv6NetworkCIDR)
 		return err
 	}
 
-	gatewayIpv6Addr, err := b.handler.ParseAddr(defaultGatewayIpv6)
+	b.dhcpConfig.GatewayIpv6 = gatewayAddr.IP.To16()
+	b.dhcpConfig.IPv6 = *vmAddr
```

Comment:
I honestly tried to touch this as little as possible since I cannot reason about any of it.

But I have to say I also don't understand why what you're proposing is better.

Reply:
I think I get what you're saying. 

What you're asking happens in the next commit; in this one, the method you're referring to doesn't even exist.

Reply:
We can ignore this since is totally addressed at follow up commits.